core/src/main/scala/org/apache/spark/io/CompressionCodec.scala

```diff
@@ -149,12 +149,7 @@ class LZFCompressionCodec(conf: SparkConf) extends CompressionCodec {
  */
 @DeveloperApi
 class SnappyCompressionCodec(conf: SparkConf) extends CompressionCodec {
-
-  try {
-    Snappy.getNativeLibraryVersion
```

Comment:
Also, I guess `getNativeLibraryVersion` might be kind of expensive due to the IO that it does, so maybe calling it less frequently will give us a perf. boost?